<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>652</width>
    <height>427</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PassKeeper</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <widget class="QWidget" name="widget_4" native="true">
      <property name="maximumSize">
       <size>
        <width>220</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLineEdit" name="searchEdit">
         <property name="whatsThis">
          <string/>
         </property>
         <property name="inputMask">
          <string notr="true"/>
         </property>
         <property name="maxLength">
          <number>100</number>
         </property>
         <property name="placeholderText">
          <string>Search for entries</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListView" name="listView">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="contextMenuPolicy">
          <enum>Qt::ActionsContextMenu</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectItems</enum>
         </property>
         <property name="selectionRectVisible">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_3" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>9</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>9</number>
          </property>
          <item>
           <widget class="QPushButton" name="addButton">
            <property name="toolTip">
             <string>Add new entry</string>
            </property>
            <property name="text">
             <string>&amp;Add</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="deleteButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Delete selected entry</string>
            </property>
            <property name="text">
             <string>&amp;Delete</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="itemWidget" native="true">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QGridLayout" name="gridLayout" columnstretch="0,0">
         <property name="sizeConstraint">
          <enum>QLayout::SetDefaultConstraint</enum>
         </property>
         <property name="horizontalSpacing">
          <number>3</number>
         </property>
         <property name="verticalSpacing">
          <number>4</number>
         </property>
         <item row="6" column="0">
          <widget class="QLCDNumber" name="PassDiff"/>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="urlEdit">
           <property name="maxLength">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_4">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>108</width>
             <height>27</height>
            </size>
           </property>
           <property name="text">
            <string>Password:</string>
           </property>
           <property name="buddy">
            <cstring>passwordEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>108</width>
             <height>27</height>
            </size>
           </property>
           <property name="text">
            <string>Website URL:</string>
           </property>
           <property name="buddy">
            <cstring>urlEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLineEdit" name="passwordEdit">
           <property name="maxLength">
            <number>50</number>
           </property>
           <property name="echoMode">
            <enum>QLineEdit::Password</enum>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="loginEdit">
           <property name="maxLength">
            <number>50</number>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QPushButton" name="copyLoginButton">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>0</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Copy user name to clipboard</string>
             </property>
             <property name="text">
              <string>Copy login</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="6" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <property name="spacing">
            <number>6</number>
           </property>
           <item>
            <widget class="QPushButton" name="copyPasswordButton">
             <property name="minimumSize">
              <size>
               <width>127</width>
               <height>0</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Copy password to clipboard</string>
             </property>
             <property name="text">
              <string>Copy password</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="showPasswordButton">
             <property name="minimumSize">
              <size>
               <width>126</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>Show Password</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="nameEdit"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Login</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Name</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="spacing">
            <number>6</number>
           </property>
           <item>
            <widget class="QPushButton" name="copyURLButton">
             <property name="toolTip">
              <string>Copy URL to clipboard</string>
             </property>
             <property name="text">
              <string>Copy URL</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="goToURLButton">
             <property name="toolTip">
              <string>Open URL in default browser</string>
             </property>
             <property name="text">
              <string>Go to URL</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Password complexity</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QWidget" name="widget_5" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="deselectButton">
            <property name="text">
             <string>Deselect</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="saveButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Save current entry</string>
            </property>
            <property name="text">
             <string>&amp;Save</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>652</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Da&amp;tabase</string>
    </property>
    <addaction name="actionChangePassword"/>
    <addaction name="actionExportDatabase"/>
    <addaction name="actionImportDatabase"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>S&amp;ettings</string>
    </property>
    <addaction name="actionPreferences"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Too&amp;ls</string>
    </property>
    <widget class="QMenu" name="menuSort_Entries">
     <property name="title">
      <string>S&amp;ort entries</string>
     </property>
     <addaction name="actionSortAscending"/>
     <addaction name="actionSortDescending"/>
    </widget>
    <addaction name="actionGeneratePassword"/>
    <addaction name="actionShowNumberedCharacters"/>
    <addaction name="menuSort_Entries"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
   <addaction name="menuSettings"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionAbout">
   <property name="text">
    <string>About PassKeeper</string>
   </property>
  </action>
  <action name="actionChangePassword">
   <property name="text">
    <string>&amp;Change password</string>
   </property>
  </action>
  <action name="actionExportDatabase">
   <property name="text">
    <string>&amp;Export database</string>
   </property>
  </action>
  <action name="actionImportDatabase">
   <property name="text">
    <string>&amp;Import database</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences</string>
   </property>
  </action>
  <action name="actionGeneratePassword">
   <property name="text">
    <string>&amp;Generate password</string>
   </property>
  </action>
  <action name="actionShowNumberedCharacters">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Show numbered characters</string>
   </property>
   <property name="toolTip">
    <string>Show password with numbered characters</string>
   </property>
   <property name="statusTip">
    <string>Show password with numbered characters</string>
   </property>
  </action>
  <action name="actionSortAscending">
   <property name="text">
    <string>Ascending order</string>
   </property>
  </action>
  <action name="actionSortDescending">
   <property name="text">
    <string>Descending order</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>listView</tabstop>
  <tabstop>addButton</tabstop>
  <tabstop>deleteButton</tabstop>
  <tabstop>searchEdit</tabstop>
  <tabstop>saveButton</tabstop>
  <tabstop>deselectButton</tabstop>
  <tabstop>urlEdit</tabstop>
  <tabstop>loginEdit</tabstop>
  <tabstop>passwordEdit</tabstop>
  <tabstop>copyURLButton</tabstop>
  <tabstop>goToURLButton</tabstop>
  <tabstop>copyLoginButton</tabstop>
  <tabstop>copyPasswordButton</tabstop>
  <tabstop>showPasswordButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>copyURLButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>copyURL()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>381</x>
     <y>123</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addnew()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nameEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>enableSaveButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>urlEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>enableSaveButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loginEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>enableSaveButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>passwordEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>enableSaveButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>579</x>
     <y>333</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>quit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGeneratePassword</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>generatePassword()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPreferences</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showPreferencesDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSortAscending</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sortAscending()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSortDescending</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sortDescending()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showPasswordButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>switchEchoMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>537</x>
     <y>241</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionImportDatabase</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>importDatabase()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExportDatabase</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>exportDatabase()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeSelectedItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>157</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>goToURLButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>goToURL()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>467</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowNumberedCharacters</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showPasswordViewer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>passwordEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>checkPassDiff()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>498</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionChangePassword</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>changePassword()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>copyLoginButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>copyLogin()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>418</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>copyPasswordButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>copyPassword()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>432</x>
     <y>241</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>searchEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>68</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>118</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>nameEdit</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>68</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>498</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addnew()</slot>
  <slot>copyURLButton()</slot>
  <slot>copyLoginButton()</slot>
  <slot>showSelectedItem()</slot>
  <slot>enableSaveButton()</slot>
  <slot>saveItem()</slot>
  <slot>quit()</slot>
  <slot>generatePassword()</slot>
  <slot>about()</slot>
  <slot>showPreferencesDialog()</slot>
  <slot>sortAscending()</slot>
  <slot>sortDescending()</slot>
  <slot>switchEchoMode()</slot>
  <slot>importDatabase()</slot>
  <slot>exportDatabase()</slot>
  <slot>removeSelectedItem()</slot>
  <slot>goToURL()</slot>
  <slot>showPasswordViewer()</slot>
  <slot>checkPassDiff()</slot>
  <slot>changePassword()</slot>
  <slot>copyLogin()</slot>
  <slot>copyPassword()</slot>
  <slot>copyURL()</slot>
 </slots>
</ui>
